### ISSUE TITLE
Beat proc fails to startup on Python 3

### ISSUE THREAD

**Issue by jaraco** [yougov/velociraptor#226]
Attempting to run vr_beat on Python 3, I get this Traceback during startup:

```
Traceback (most recent call last):
  File "/app/.heroku/venv/lib/python3.6/site-packages/django/core/cache/__init__.py", line 84, in _create_cache
    backend_cls = import_string(backend)
  File "/app/.heroku/venv/lib/python3.6/site-packages/django/utils/module_loading.py", line 26, in import_string
    module = import_module(module_path)
  File "/usr/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/app/.heroku/venv/lib/python3.6/site-packages/redis_cache/__init__.py", line 1, in <module>
    from redis_cache.cache import RedisCache
  File "/app/.heroku/venv/lib/python3.6/site-packages/redis_cache/cache.py", line 4, in <module>
    from django.utils.encoding import smart_unicode, smart_str
ImportError: cannot import name 'smart_unicode'
```

**Comment by jaraco** [comment]
This issue was fixed in 2013: https://github.com/sebleier/django-redis-cache/commit/59caf9cda79aef8b29308e1f484e8c3a27f31181#diff-ade4a288c4a78ecd3fdf0954ed19b749

**Comment by lbolla** [comment]
Unfortunately, bumping `django-redis-cache` is not enough. We depend on https://bitbucket.org/btubbs/celery-schedulers which is unmaintained since 2016 and works on Py2 only. I'll fork it, add it to the yougov account and try to make it py3 compatible. I wish I could volunteer @btubbs to do it!

**Comment by lbolla** [comment]
@jaraco with the above fixes to `celery-schedulers` `vr_beat` now starts in Python3. How do you want to proceed to release YouGov's fork of `celery-schedulers`? Maybe release it under a different name (`yg.celery-schedulers`)?

**Comment by jaraco** [comment]
I was afraid that it wasn't as simple as bumping one dependency. Good progress though!

For celery-schedulers (and other dependencies that have defects), here's what I recommend:

(å) Unpin the dependency in vr.server so later versions may be accepted.
(a) fork the project and fix the defect.
(b) send a pull request... but since many of these projects are abandoned or slowly maintained, you probably can't expect the PR to be merged or released in any reasonable time frame, so...
(c) In the fork, create a branch `yougov` against the last stable release and add/amend `setup.cfg` with `egg_info.tag_build=.yg1`. This creates a branch from which to create private post releases. Merge or cherry-pick the changes you need from the PR (and any other changes you wish to incorporate in this private post-release). Tag the changes with `git tag -m tag $(python setup.py --version)`.
(d) Use `devpi login $USER` to log into our internal repository and cut the release using `rm -rf dist/*; python setup.py sdist; devpi upload dist/*` to publish the patched version privately.
(e) In our deployment repo, pin to `(last-release).yg1` for the dependency, referencing the PR above as the reason.
(f) If you have subsequent changes to make to the upstream project, repeat the process above, merging into the `yougov` branch and bumping the `egg_info.tag_build=.ygN` to the next value.

**Comment by lbolla** [comment]
I've released `celery-schedulers==0.0.4` which is Python3 compatible.
